tests/a11y-no-visually-hidden-interactive-element.js

```diff
@@ -0,0 +1,45 @@
+const rule = require('../lib/rules/a11y-no-visually-hidden-interactive-element')
+const RuleTester = require('eslint').RuleTester
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 'latest',
+    sourceType: 'module',
+    ecmaFeatures: {
+      jsx: true,
+    },
+  },
+})
+
+const errorMessage =
+  'Avoid visually hidding interactive elements. Visually hiding interactive elements can be confusing to sighted keyboard users as it appears their focus has been lost when they navigate to the hidden element.'
+
+ruleTester.run('a11y-no-visually-hidden-interactive-element', rule, {
+  valid: [
+    {code: '<VisuallyHidden as="h2">Submit</VisuallyHidden>'},
+    {code: "<div className='sr-only'>Text</div>;"},
+    {code: '<VisuallyHidden><div>Text</div></VisuallyHidden>'},
+    {code: "<div className='other visually-hidden'>Text</div>;"},
+    {code: "<span className='sr-only'>Text</span>;"},
+    {code: "<button className='other'>Submit</button>"},
+    {code: "<input className='sr-only' />"},
+    {code: "<a className='other show-on-focus'>skip to main content</a>"},
```

Comment:
Check that `show-on-focus` class is valid! This way I don't have to remove the `a` tag from the list.